src/plugins/kibana_react/public/core/types.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CoreSetup, CoreStart } from '../../../../core/public';
+
+export { CoreSetup, CoreStart };
+
+export type MaybePromise<T> = T | Promise<T>;
+export type NPPluginLifecycle<Core, Imports, Exports> = (
+  core: Core,
+  plugins: Imports
+) => MaybePromise<Exports>;
+
+export interface NPClientPlugin<
```

Comment:
Does this file need to be exporting anything besides `Core` and `KibanaReactContextValue`?

I'm assuming the others are just here for convenience, but I think we should consider removing them unless we have multiple concrete places where we are certain they would be used. We can always add more things later; it's much harder to remove 😉

Reply:
You're right, removed.